File: sample/src/main/java/com/iktwo/sample/VIPersonDelegate.java
package com.iktwo.sample;

import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.iktwo.binder.BindingViewHolder;
import com.iktwo.binder.ViewDelegate;
import com.iktwo.sample.databinding.DelegateVipBinding;

public class VIPersonDelegate extends ViewDelegate<BindingViewHolder<DelegateVipBinding>> {


    public VIPersonDelegate() {

    }

    @Override
    public int getItemViewType() {
        return 1;
    }

    @Override
    public BindingViewHolder<DelegateVipBinding> onCreateViewHolder(ViewGroup viewGroup) {
        DelegateVipBinding binding = DataBindingUtil.inflate(LayoutInflater.from(viewGroup.getContext()),
                R.layout.delegate_vip, viewGroup, false);

        return new BindingViewHolder<>(binding);
    }

    @Override
    public void onBindViewHolder(BindingViewHolder<DelegateVipBinding> viewHolder, Object item, int position) {
        viewHolder.getBinding().setModel((Person) item);
        viewHolder.getBinding().executePendingBindings();
    }

    @Override
    public boolean canHandleItemTypeAtPosition(@NonNull Object item, int position) {
        if (item instanceof Person) {
            return ((Person) item).isVI;
        }

        return false;
    }
}

File: sample/src/main/java/com/iktwo/sample/VIPersonDelegate.kt
package com.iktwo.sample

import android.databinding.DataBindingUtil
import android.view.LayoutInflater
import android.view.ViewGroup

import com.iktwo.binder.BindingViewHolder
import com.iktwo.binder.ViewDelegate
import com.iktwo.sample.databinding.DelegateVipBinding

class VIPersonDelegate(private val handler: PersonDelegateHandler) : ViewDelegate<BindingViewHolder<DelegateVipBinding>>() {

    override val itemViewType: Int
        get() = 1

    override fun onCreateViewHolder(viewGroup: ViewGroup): BindingViewHolder<DelegateVipBinding> {
        val binding = DataBindingUtil.inflate<DelegateVipBinding>(LayoutInflater.from(viewGroup.context),
                R.layout.delegate_vip, viewGroup, false)

        return BindingViewHolder(binding)
    }

    override fun onBindViewHolder(viewHolder: BindingViewHolder<DelegateVipBinding>, item: Any, position: Int) {
        viewHolder.binding.model = item as Person
        viewHolder.binding.executePendingBindings()
    }

    override fun canHandleItemTypeAtPosition(item: Any, position: Int): Boolean {
        return (item as? Person)?.isVI ?: false

    }
}
